### DISCUSSION TITLE
Getting window in ssr

### DISCUSSION THREAD

**Discussion by jamiviz** [vercel/next.js#41530] [Help]
Hello , i am facing a problem with the new version of Nextjs v13 to make window work with Ssr.

The idea is if the (window.innerWidth >= 768) the server should render "/desktop_bg.jpg"  , and if (window.innerWidth < 768) the server should render  "/mobile_bg.jpg".
But the problem that (window.innerWidth) isnt accessible in the ssr .

My code : 

`
  const [bg, setBg] = useState("");

  const changeBg = () => {
    if (window.innerWidth >= 768) {
      return setBg("/desktop_bg.jpg");
    }
    if (window.innerWidth < 768) {
      return setBg("/mobile_bg.jpg");
    }
  };

  useEffect(() => {
    if (typeof window !== "undefined") {
      window.addEventListener("resize", changeBg);
      return () => window.removeEventListener("resize", changeBg);
    }
  });
`

Thanks

**Comment by icyJoseph** [comment]
Hi,

This is a common pitfall of switching from CSR to SSR.

```tsx
import type { NextPage } from "next";
import { useEffect, useState } from "react";

const Home: NextPage = () => {
  const [bg, setBg] = useState("");


  useEffect(() => { 
    const changeBg = () => {
      return setBg(window.innerWidth >= 768 ? "/desktop_bg.jpg" : "/mobile_bg.jpg");
    };
    // inside useEffect the window is always defined!
    window.addEventListener("resize", changeBg);

    changeBg(); // as soon as you are on the client, run this handler!

    return () => window.removeEventListener("resize", changeBg);
  }, []);

  return (
    <div>
      <span>{bg}</span>
    </div>
  );
};

export default Home;

```

A couple of things here:

- You should do this with CSS instead
- If I make a request as a bot I'll not see `bg` ever change 
- Don't forget the dependency array
- it's better to define the handler inside use-effect, to prevent an additional dependency

You might want to read: 
- https://gist.github.com/gaearon/e7d97cdf38a2907924ea12e4ebdf3c85
- and https://medium.com/@alexandereardon/uselayouteffect-and-ssr-192986cdcf7a
